test/test_mp_all_gather.py

```diff
@@ -57,6 +58,48 @@ def _mp_fn(index):
           f'Failed to create two replica groups with {world_size} replicas',
           file=sys.stderr)
 
+    # Testing with a single replica group and tensor list as input
+    ordinal_tensors = [
+        torch.tensor([i * 1000 + index], dtype=torch.float).to(device)
+        for i in range(input_list_size)
+    ]
+    # TODO: add support for list input with pin_layout=True and output=None
+    result_list = xm.all_gather(ordinal_tensors, dim=0, pin_layout=False)
+
+    for i, result in enumerate(result_list):
+      cpu_result = result.cpu()
+      expected = i * 1000 + torch.arange(world_size, dtype=torch.float)
+      if not cpu_result.allclose(expected):
+        print(
+            'xm.all_gather() produced wrong reductions for item {i} in result list',
+            file=sys.stderr)
+        print(f'[{index}] {cpu_result}', file=sys.stderr)
+        sys.exit(1)
+
+    # Testing with a single replica group and tensor list as input and output!=None (out-of-place)
+    ordinal_tensors = [
```

Comment:
question, couldn't we just reuse the `ordinal_tensors` from the previous test case? I thought `all_gather` won't touch the original tensor.

Reply:
You are right. Fixed in latest commit.